api/envoy/api/v2/cds.proto

```diff
@@ -531,6 +531,14 @@ message Cluster {
   // If this flag is not set to true, Envoy will wait until the hosts fail active health
   // checking before removing it from the cluster.
   bool drain_connections_on_host_removal = 32;
+
```

Comment:
I'm not sure about what I did here. I did `#not-implemented-hide `for src_transparent since it won't do anything useful. Likewise, since mark isn't fully tested out in an end-to-end feature, I decide to use `#proto-status: draft` for now.

Reply:
We may also want to discuss the format of src_transparent. Should it be a bool, or itself a separate struct? I'm thinking of extensibility, and the fact we were considering adding some options like how to handle failure to bind.

Would we make a struct along the following lines?
```
{
bool src_transparent
enum bind_failure_strategy
{
...
}
}
```
Or would it make sense to keep it src_transparent separate as it is, then add a separate struct which contains options that are relevant when src_transparent is true?

Reply:
Regarding the flags, I'm not sure yet. I'll probably know for sure in the next few weeks. 

I'll verify the mark functionality in the next few weeks as well.

Reply:
Because I didn't notice it? :) I think it makes more sense to put them there for sure.

Reply:
Moved into UpstreamConnectionOptions.